File: java/InsertionSort.java

import java.util.Scanner;

public class InsertionSort {
	public static void main(String[] args) {

		int N = 20;
		int vetor[] = new int[N];
		int aux;
		Scanner entrada = new Scanner(System.in);

		for (int i = 0; i < N; i++) {
			System.out.print("Digite o " + (i + 1) + "° número: ");
			vetor[i] = entrada.nextInt();
		}

		int j = 0;
		for (int i = 1; i < N; i++) {
			aux = vetor[i];
			j = i - 1;
			while (j >= 0 && vetor[j] > aux) {
				vetor[j+1] = vetor[j];
				j--;
			}
			vetor[j+1] = aux;
		}

		for (int i = 0; i < N; i++) {
			System.out.println((i + 1) + "° número: " + vetor[i]);
		}
	}
}

File: C++/insertion_sort.cpp
#include<iostream>

using namespace std;

int main() {
	int aux, N = 5;
	int vetor[N];

	for (int i = 0; i < N; i++) {
		cout << "Digite o " << i + 1 << "° número: ";
		cin >> vetor[i];
	}

	int j = 0;
	for (int i = 1; i < N; i++) {
		aux = vetor[i];
		j = i - 1;
		while (j >= 0 && vetor[j] > aux) {
			vetor[j+1] = vetor[j];
			j--;
		}
		vetor[j+1] = aux;
	}


	for (int i = 0; i < N; i++) {
		cout << (i + 1) << "° número: " << vetor[i] << "\n";
	}

	return 0;
}
